### PR TITLE
Update .gitignore - adds py environment folder

### PR DESCRIPTION
**Pull request by raul-marquez-csa**
### Update .gitignore - adds py environment folder

All the files and folders in the py environment folder are appearing as new changes to commit. This repository doesn't include that folder, so I added the py folder to ignore it.

This command creates the py environment folder:
`./scripts/build_python.sh -i py`

Explanation is here:
https://github.com/project-chip/connectedhomeip/blob/bf4edc4f0eaf949de3e39082ec4b8e64122cfd67/docs/testing/python.md?plain=1#L517-L523

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -85,3 +85,5 @@ examples/*/esp32/dependencies.lock
 # jupyter temporary files
 .ipynb_checkpoints
 
+# py environment folder
+py
```
Comment (tcarmelveilleux) [inline .gitignore:89]:
```suggestion
py
__pycache__
```

**Review by tcarmelveilleux — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (raul-marquez-csa) [reply .gitignore:89]: That's interesting, __pycache__ was already in the file, add it again?

**Review by woody-apple — APPROVED** [review]

**Review by andy31415 — CHANGES_REQUESTED** [review]
What is a `py` environment folder? What build process creates that?

I do not have such a thing.

**Comment by andy31415** [comment]
> What is a `py` environment folder? What build process creates that?
> 
> I do not have such a thing.

Specifcally:
  - name does not feel special (no `__` prefix nor `.` even for hidden)
  - name is actually common, like `scripts/tests/py/` and `src/test_driver/efr32/py/`
  
We have to figure out what build command creates it and dirties the source tree. Please add this information to the PR summary so we can determine if the command is buggy or if we really need to ignore (and add details to gitignore).

Just ignoring things that dirty our source tree is not ok (we should not accept dirtying of the source tree)

**Comment by raul-marquez-csa** [comment]
> > What is a `py` environment folder? What build process creates that?
> > I do not have such a thing.
> 
> Specifcally:
> 
> * name does not feel special (no `__` prefix nor `.` even for hidden)
> * name is actually common, like `scripts/tests/py/` and `src/test_driver/efr32/py/`
> 
> We have to figure out what build command creates it and dirties the source tree. Please add this information to the PR summary so we can determine if the command is buggy or if we really need to ignore (and add details to gitignore).
> 
> Just ignoring things that dirty our source tree is not ok (we should not accept dirtying of the source tree)

This command creates it
`./scripts/build_python.sh -i py`

Explanation is here:
https://github.com/project-chip/connectedhomeip/blob/bf4edc4f0eaf949de3e39082ec4b8e64122cfd67/docs/testing/python.md?plain=1#L517-L523

**Comment by raul-marquez-csa** [comment]
I can update it to this
`/py`
This will only target the py folder in the root folder, not subfolders

**Comment by bzbarsky-apple** [comment]
Shouldn't builds go into `out` or something like that which already gets gitignored?

**Comment by andy31415** [comment]
> > > What is a `py` environment folder? What build process creates that?
> > > I do not have such a thing.
> > 
> > 
> > Specifcally:
> > 
> > * name does not feel special (no `__` prefix nor `.` even for hidden)
> > * name is actually common, like `scripts/tests/py/` and `src/test_driver/efr32/py/`
> > 
> > We have to figure out what build command creates it and dirties the source tree. Please add this information to the PR summary so we can determine if the command is buggy or if we really need to ignore (and add details to gitignore).
> > Just ignoring things that dirty our source tree is not ok (we should not accept dirtying of the source tree)
> 
> This command creates it `./scripts/build_python.sh -i py`
> 
> Explanation is here:
> 
> https://github.com/project-chip/connectedhomeip/blob/bf4edc4f0eaf949de3e39082ec4b8e64122cfd67/docs/testing/python.md?plain=1#L517-L523

We should fix the documentation to place it in `out/py`.

**Comment by andy31415** [comment]
Closing old PR: we have in the mean time updated our documentation in a different PR (#35901)

**Pull request closed without merging**